cmd/kubelet/app/options/options.go

```diff
@@ -409,6 +424,8 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 		mainfs.AddFlagSet(fs)
 	}()
 
+	// Instance specific flags
+	fs.BoolVar(&c.RegisterNode, "register-node", c.RegisterNode, "Register the node with the apiserver. If --kubeconfig is not provided, this flag is irrelevant, as the Kubelet won't have an apiserver to register with.")
```

Comment:
That'll also force us to deal with parsing flags into the instance config that don't map to the shared config.

Reply:
Having the flags only in the instance-specific will make Kubelet read from different Internal structs (instance/shared).
Given that even if they are instance-specific, still makes sense to keep then in the shared as well. Otherwise, it is necessary to extend the `KubeletServer` to hold both pointers and consequently refactor

Reply:
`--hostname-override` and `--node-ip` never make sense to set via shared config, so I'd prefer to avoid including them in that object (especially given dynamic kubelet config).

Reply:
sure, makes sense.